packages/angular_devkit/build_angular/src/angular-cli-files/models/webpack-configs/utils.ts

```diff
@@ -66,3 +67,25 @@ export function normalizeExtraEntryPoints(
     return normalizedEntry;
   })
 }
+
+export function getSourceMapDevTool(
+  scriptsSourceMap: boolean,
+  stylesSourceMap: boolean,
+  hiddenSourceMap = false,
+  inlineSourceMap = false,
+): SourceMapDevToolPlugin {
+  const include = [];
+  if (scriptsSourceMap) {
+    include.push(/js$/);
+  }
+
+  if (stylesSourceMap) {
+    include.push(/css$/);
+  }
+
+  return new SourceMapDevToolPlugin({
+    filename: inlineSourceMap ? undefined : '[file].map',
```

Comment:
does the file format account for the hashing in production?

Reply:
How about ill add a test for that? :)